Portal/vip-portal/src/main/resources/vm/wrapper.vm

```diff
@@ -79,107 +91,41 @@ fi
-#set($ob="{")
-#foreach($outputFile in $tool.getOutputFiles())
-## Replace command-line keys
-#if($outputFile.getCommandLineKey())
-	#if($outputFile.getCommandLineFlag() && $outputFile.getCommandLineFlag() != "")
-		#set($newValue="""$outputFile.getCommandLineFlag() $$outputFile.getId().toUpperCase()""")
-	#else
-		#set($newValue="""$$outputFile.getId().toUpperCase()""")
-	#end
-BOUTIQUES_COMMAND_LINE=$${ob}BOUTIQUES_COMMAND_LINE//'$outputFile.getCommandLineKey()'/$newValue}
-#end
-#end 
-
-#foreach($input in $tool.getInputs())
-  #if($input.isOptional())
-if [ "$$input.getId().toUpperCase()" != "no" ] && [ "$$input.getId().toUpperCase()" != "No_value_provided" ]
-then
-  # Optional input is active.
-  #end
-  #if($input.getType() != "Flag")
-	#if($input.getCommandLineFlag() && $input.getCommandLineFlag() != "")
-		#set($newValue="""$input.getCommandLineFlag() $$input.getId().toUpperCase()""")
-	#else
-		#set($newValue="""$$input.getId().toUpperCase()""")
-	#end
-    #if($input.getType() == "File" && $input.isOptional())
-  # Optional input is active and it is a file. The file needs to be downloaded (optional files are not handled by GASW).
-  downloadLFN $$input.getId().toUpperCase()
-  LOCAL_FILE_NAME=`basename $$input.getId().toUpperCase()`
-		#if($input.getCommandLineFlag() && $input.getCommandLineFlag() != "")
-			BOUTIQUES_COMMAND_LINE=$${ob}BOUTIQUES_COMMAND_LINE//'$input.getCommandLineKey()'/"$input.getCommandLineFlag() ${LOCAL_FILE_NAME}"}
-		#else
-			BOUTIQUES_COMMAND_LINE=$${ob}BOUTIQUES_COMMAND_LINE//'$input.getCommandLineKey()'/"${LOCAL_FILE_NAME}"}
-		#end
-    #else
-  BOUTIQUES_COMMAND_LINE=$${ob}BOUTIQUES_COMMAND_LINE//'$input.getCommandLineKey()'/$newValue}
-    #end
-  #else
-if [ "$$input.getId().toUpperCase()" = "true" ]
-then
-  # flag is set: replace command-line key by flag value
-  BOUTIQUES_COMMAND_LINE=$${ob}BOUTIQUES_COMMAND_LINE//'$input.getCommandLineKey()'/$input.getCommandLineFlag()}
-else
-  # flag is unset: remove command-line key from command-line. 
-  BOUTIQUES_COMMAND_LINE=$${ob}BOUTIQUES_COMMAND_LINE//'$input.getCommandLineKey()'/""}
-fi
-#end
-#if($input.isOptional())
-else
-  # Optional input is not active: remove command-line key from command-line. 
-  BOUTIQUES_COMMAND_LINE=$${ob}BOUTIQUES_COMMAND_LINE//'$input.getCommandLineKey()'/""}
-fi
-#end
-#end
+cat << JSONPARAMETERS  >> input_param_file.json
+}
+JSONPARAMETERS
 
 ##########################
 # Command-line execution #
 ##########################
 
-#if($tool.getDockerImage())
-cat << DOCKERJOB > .dockerjob.sh
-#!/bin/bash -l
-${BOUTIQUES_COMMAND_LINE}
-DOCKERJOB
-
-chmod 755 .dockerjob.sh
-docker pull $tool.getDockerImage()
-docker run --rm -v $PWD:/gasw-execution-dir -v $PWD/../cache:$PWD/../cache -w /gasw-execution-dir -u `id -u`:`id -g` $tool.getDockerImage()  ./.dockerjob.sh 
-#else
-${BOUTIQUES_COMMAND_LINE}
-#end
+TOOLNAME="$tool.getName()"
+JSONFILE="${TOOLNAME}.json"
+
+checkBosh
```

Comment:
You could put your boutiques installation code here instead of having a function somewhere else, for consistency.